runelite-client/src/main/java/net/runelite/client/plugins/herbsackpricecheck/HerbSackPriceCheckPlugin.java

```diff
@@ -0,0 +1,175 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.herbsackpricecheck;
+
+import net.runelite.client.eventbus.Subscribe;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.MenuOptionClicked;
+import net.runelite.client.chat.ChatColorType;
+import net.runelite.client.chat.ChatMessageManager;
+import net.runelite.client.chat.QueuedMessage;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.util.StackFormatter;
+import net.runelite.http.api.item.Item;
+import net.runelite.http.api.item.ItemPrice;
+import net.runelite.http.api.item.SearchResult;
+import net.runelite.client.chat.ChatMessageBuilder;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import static net.runelite.api.ItemID.HERB_SACK;
+
+@PluginDescriptor(
+		name = "Herb sack price check",
+		description = "Price checks the herbs in herb sack",
+		tags = {"herbs", "prices"}
+)
+
+@Slf4j
+public class HerbSackPriceCheckPlugin extends Plugin
+{
+	private static boolean gettingHerbs = false;
+	private static ArrayList<ChatMessage> herbsInChatMessage = new ArrayList<>();
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	private ChatMessageManager chatMessageManager;
+
+	@Inject
+	private Client client;
+
+	@Subscribe
+	private void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (!event.getMenuOption().equals("Check") || event.getId() != HERB_SACK)
+		{
+			return;
+		}
+		gettingHerbs = true;
+	}
+
+	@Subscribe
+	private void onChatMessage(ChatMessage message)
+	{
+		if (message.getType() != ChatMessageType.GAMEMESSAGE)
+			return;
+
+		if (message.getMessage().contains("x Grimy") && gettingHerbs)
+		{
+			herbsInChatMessage.add(message);
+		}
+	}
+
+	@Subscribe
+	private void onGameTick(GameTick event)
+	{
+		if (gettingHerbs && !herbsInChatMessage.isEmpty())
+		{
+			HashMap<String, Integer> herbsWithQuantity = reformatHerbChatMessages();
+			int totalValue = herbPriceLookup(herbsWithQuantity);
+
+			if (totalValue != -1)
+			{
+				buildValueMessage(totalValue);
+			}
+
+			gettingHerbs = false;
```

Comment:
The values of these fields should be reset upon plugin disable as well, so these lines which reset their values can be moved to a helper function.